extensions/interactions/ItemSelectionInput/directives/oppia-response-item-selection-input.component.spec.ts

```diff
@@ -65,4 +65,11 @@ describe('ResponseItemSelectionInput', () => {
 
     expect(component.responses).toEqual(['choice 1']);
   });
+  it('should set the answers and responses arrays to empty when no answer ' +
+    'is selected', () => {
+    component.answer = '""';
```

Comment:
I'm not sure I understand what's going on here.

Firstly, this doesn't seem like an empty answer?

Secondly, the test says that it should set the answers array to empty, but there's no assertion for that here.

Thirdly, there's no pre-check for the test to ensure that emptiness isn't just due to how things were setup initially (i.e. the background conditions in the test), rather than due to the code actually working.

Reply:
> Firstly, this doesn't seem like an empty answer?

I've used this to represent no answer being selected because, upon logging some values, I discovered that this is what is passed to the `JSON.parse()` function when the submit button is clicked with no answer selected by the learner, which is what I want to emulate in this test.

> Secondly, the test says that it should set the answers array to empty, but there's no assertion for that here.

The answer variable that the test refers to is a local variable and not a property of the component class, and can't be accessed outside of it. But the assertion for the responses property of the class suffices because it is directly built using the answer array in question. An empty responses array means an empty answers array.

Of course, I could make the answer variable the test is referring to a property of the class so that it can be accessed outside, if you're suggesting that checking the responses array only is not enough.

> Thirdly, there's no pre-check for the test to ensure that emptiness isn't just due to how things were setup initially (i.e. the background conditions in the test), rather than due to the code actually working.

You're right I missed this, the responses array is meant to be undefined before component initialisation -- I'll add a check for the same

Reply:
Done, thanks for pointing out.

Reply:
> I've used this to represent no answer being selected because, upon logging some values, I discovered that this is what is passed to the JSON.parse() function when the submit button is clicked with no answer selected by the learner, which is what I want to emulate in this test.

Ah ok. Fair enough, but please leave a comment in the code, since that's not clear.

> Of course, I could make the answer variable the test is referring to, a property of the class so that it can be accessed outside, if you're suggesting that checking the responses array only is not enough.

I would suggest changing the test name instead, since the test isn't testing all of the things it currently says it is.